src/tools/android/java/com/google/devtools/build/android/AndroidResourceProcessor.java

```diff
@@ -343,24 +343,17 @@ public MergedAndroidData processResources(
           dependencyData, customPackageForR, androidManifest, sourceOut);
     }
     // Reset the output date stamps.
-    if (proguardOut != null) {
-      Files.setLastModifiedTime(proguardOut, FileTime.fromMillis(0L));
-    }
-    if (mainDexProguardOut != null) {
-      Files.setLastModifiedTime(mainDexProguardOut, FileTime.fromMillis(0L));
-    }
     if (packageOut != null) {
-      Files.setLastModifiedTime(packageOut, FileTime.fromMillis(0L));
       if (!splits.isEmpty()) {
         Iterable<Path> splitFilenames = findAndRenameSplitPackages(packageOut, splits);
         for (Path splitFilename : splitFilenames) {
-          Files.setLastModifiedTime(splitFilename, FileTime.fromMillis(0L));
+          // Iterate over the elements to get the side-effects of findAndRenameSplitPackages.
+          // TODO(ajmichael): find out if findAndRenameSplitPackages's side-effects are actually
```

Comment:
The side-effects are certainly needed, however we don't need to actually iterate, the return is an ImmutableList. I can clean this up in a follow-up.

Reply:
Good point, and thanks for confirming it! I'll push a new commit that renames the method and removes the TODO.